### PR TITLE
contrib/go-redis/redis.v7: support for redis v7 package

### LINKED ISSUE #857: proposal: add support for go-redis v7
`go-redis` v6 and v8 is supported but v7 is not. I have a few dependencies on v7 so I want to add support for it. Please share what you think.
Discussion on v7 was originally in #606

### PR DESCRIPTION
**Pull request by nicoledanuwidjaja**
Add support for go-redis v7. Used go-redis v8 as a model and adjusted methods accordingly to account for Context usage.

Fixes #857.

### CODE REVIEW

**Comment by gbbr** [comment]
Your second commit still contains the entire folder, and not only the changes. It will be nice to have two commits:
* Commit 1: copies over the other integration
* Commit 2: updates the import path and interface to match v7

File: contrib/go-redis/redis.v7/example_test.go [hunk 1]
```diff
@@ -9,39 +9,40 @@ import (
...
 	opts := &redis.Options{Addr: "127.0.0.1", Password: "", DB: 0}
 	c := redistrace.NewClient(opts)
 
 	// any action emits a span
-	c.Set(ctx, "test_key", "test_value", 0)
+	c.Set("test_key", "test_value", 0)
 
 	// optionally, create a new root span
-	root, ctx := tracer.StartSpanFromContext(context.Background(), "parent.request",
+	root, _ := tracer.StartSpanFromContext(context.Background(), "parent.request",
```
Comment (gbbr) [inline contrib/go-redis/redis.v7/example_test.go:29]: If we're not using a context, there is no need to use `tracer.StartSpanFromContext`. Just use `tracer.StartSpan`

File: contrib/go-redis/redis.v7/example_test.go [hunk 2]
```diff
@@ -9,39 +9,40 @@ import (
...
 		tracer.SpanType(ext.SpanTypeRedis),
 		tracer.ServiceName("web"),
 		tracer.ResourceName("/home"),
 	)
 
+	// set the context on the client
```
Comment (gbbr) [inline contrib/go-redis/redis.v7/example_test.go:35]: We shouldn't leave commented out code anywhere.

File: contrib/go-redis/redis.v7/option.go [hunk 3]
```diff
@@ -11,6 +11,10 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/internal"
 )
 
+const (
```
Comment (gbbr) [inline contrib/go-redis/redis.v7/option.go:14]: Why did you add this constant? We don't really need a global constant. It's not reused anywhere. Best revert back to how it was...

File: contrib/go-redis/redis.v7/redis.go [hunk 4]
```diff
@@ -10,17 +10,15 @@ package redis
 import (
 	"bytes"
 	"context"
-
 	"math"
 	"net"
 	"strconv"
 	"strings"
 
+	"github.com/go-redis/redis/v7"
```
Comment (gbbr) [inline contrib/go-redis/redis.v7/redis.go:18]: This should go back down where it was, separated by a blank line.

File: contrib/go-redis/redis.v7/redis_test.go [hunk 5]
```diff
@@ -277,24 +272,23 @@ func TestChildSpan(t *testing.T) {
 	}
 	assert.NotNil(parent)
 	assert.NotNil(child)
-
+	assert.Equal(ctx, client.Context())
```
Comment (gbbr) [inline contrib/go-redis/redis.v7/redis_test.go:275]: This assertion is not needed. This is the responsibility of the upstream library to test.

File: contrib/go-redis/redis.v7/redis_test.go [hunk 6]
```diff
@@ -249,25 +245,24 @@ func TestPipeline(t *testing.T) {
...
 	mt := mocktracer.Start()
 	defer mt.Stop()
 
 	// Parent span
 	client := NewClient(opts, WithServiceName("my-redis"))
-	root, ctx := tracer.StartSpanFromContext(ctx, "parent.span")
-
-	client.Set(ctx, "test_key", "test_value", 0)
+	root, ctx := tracer.StartSpanFromContext(context.Background(), "parent.span")
+	client = client.WithContext(ctx)
```
Comment (gbbr) [inline contrib/go-redis/redis.v7/redis_test.go:256]: 👍

File: contrib/go-redis/redis.v7/redis_test.go [hunk 7]
```diff
@@ -457,6 +447,8 @@ func TestWithContext(t *testing.T) {
 		}
 	}
 
+	assert.Equal(ctx1, client1.Context())
+	assert.Equal(ctx2, client2.Context())
```
Comment (gbbr) [inline contrib/go-redis/redis.v7/redis_test.go:451]: Same here. We don't really have to test this, since these methods are not implemented by us. You're testing that the child/parent relationship is ok and that's all that's needed.

**Review by gbbr — COMMENTED** [review]
Nice! Left a few nits

File: contrib/go-redis/redis.v7/example_test.go (hunk 2 above)
Comment (nicoledanuwidjaja) [reply contrib/go-redis/redis.v7/example_test.go:35]: Uncommented it

**Comment by gbbr** [comment]
Can you please fix the commits as specified in https://github.com/DataDog/dd-trace-go/pull/862#issuecomment-792608716 ?

**Review by gbbr — APPROVED** [review]
LGTM! Thanks a lot for the effort.
